### PR TITLE
refactor: Replace require with import statement

### PR DESCRIPTION
**Pull request by dplewis**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/Parse-SDK-JS/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/Parse-SDK-JS/blob/alpha/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/Parse-SDK-JS/issues?q=is%3Aissue).

## Issue

The compilers cannot optimize the imports and remove unused code.

See discussion https://github.com/parse-community/Parse-SDK-JS/issues/1399

## Approach

The only files using `require` are as follows:
CryptoController - Babel doesn't strip out unused imports
EventEmitter - Babel doesn't strip out unused imports
ParseFile - Can't use ES6 modules like https and http
RESTController - xmlhttprequest node package can't be imported
CoreManager - Can't get package version from package.json using import

Convert the following files to TS
```
Analytics
AnonymousUtils
CloudCode
CryptoController
EventEmitter
LocalDatastoreController
OfflineQuery
RESTController
Scoket.weapp.js
Storage
StorageController
StorageController.browser
StorageController.default
StorageController.react-native
StorageController.weapp
WebSocketController
Xhr.weapp
```

### CODE REVIEW

File: src/ParseFile.ts [hunk 1]
```diff
@@ -487,7 +485,7 @@ const DefaultController = {
       return this.downloadAjax(uri, options);
     } else if (process.env.PARSE_BUILD === 'node') {
       return new Promise((resolve, reject) => {
-        const client = uri.indexOf('https') === 0 ? https : http;
+        const client = uri.indexOf('https') === 0 ? require('https') : require('http');
```
Comment (mtrezza) [inline src/ParseFile.ts]: Why not import?

File: src/RESTController.ts [hunk 2]
```diff
@@ -46,7 +45,7 @@ if (typeof XMLHttpRequest !== 'undefined') {
   XHR = XMLHttpRequest;
 }
 if (process.env.PARSE_BUILD === 'node') {
-  XHR = xmlhttprequest.XMLHttpRequest;
+  XHR = require('xmlhttprequest').XMLHttpRequest;
```
Comment (mtrezza) [inline src/RESTController.ts:48]: Why not import?

**Review by mtrezza — APPROVED** [review]

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.2.0-alpha.2](https://github.com/parse-community/Parse-SDK-JS/releases/tag/5.2.0-alpha.2)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.2.0-beta.1](https://github.com/parse-community/Parse-SDK-JS/releases/tag/5.2.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.2.0](https://github.com/parse-community/Parse-SDK-JS/releases/tag/5.2.0)

**Pull request merged**